allo/backend/hls.py

```diff
@@ -168,6 +170,39 @@ def __init__(
             allo_d.register_dialect(ctx)
             self.module = Module.parse(str(mod), ctx)
             self.func = find_func_in_module(self.module, top_func_name)
+
+            if platform == "ihls":
+                assert mode in {
+                    "fpga_emulator",
+                    "source_file_only",
+                    None,
+                }, "Invalid mode"
+
+                if project is not None:
+                    filename = f"{project}"
+                else:
+                    raise RuntimeError(
+                        "Error: if platfrom is ihls, 'project' argument must not be None."
+                    )
+
+                if mode == "fpga_emulator":
+                    result = subprocess.run(
+                        [
+                            f" icpx -fintelfpga -DFPGA_EMULATOR .//{filename} -o {filename}.exe"
+                        ],
+                        capture_output=True,
+                        text=True,
+                        check=False,
+                    )
+                    print(result.stdout)
+                elif mode == "source_file_only":
+                    print(
+                        f"Generated Intel HLS source file kernel.cpp has been created successfully in your current directory under '{filename}' folder."
+                    )
+                    print(
+                        "mode has been set to source_file_only, the output will only be the souce intel HLS code"
+                    )
```

Comment:
I don't think we need this mode. Users can always access the source code from the built folder